gradle.properties

```diff
@@ -59,57 +60,74 @@ ZCASH_RELEASE_KEYSTORE_PASSWORD=
 ZCASH_RELEASE_KEY_ALIAS=
 ZCASH_RELEASE_KEY_ALIAS_PASSWORD=
 
+# Switch this property to true only if you need to sign the release build with a debug key for local development.
+IS_SIGN_RELEASE_BUILD_WITH_DEBUG_KEY=false
+
 # Versions
-ANDROID_MIN_SDK_VERSION=19
+ANDROID_MIN_SDK_VERSION=21
+ANDROID_MIN_BENCHMARK_VERSION=24
 ANDROID_TARGET_SDK_VERSION=33
 ANDROID_COMPILE_SDK_VERSION=33
 
+# TODO[#317]: Update NDK to 24.0.7856742
 # TODO[#317]: https://github.com/zcash/zcash-android-wallet-sdk/issues/317
 # When changing this, be sure to update .github/actions/setup/action.yml
 ANDROID_NDK_VERSION=22.1.7171670
 
-ANDROID_GRADLE_PLUGIN_VERSION=7.2.2
-DETEKT_VERSION=1.21.0
-DOKKA_VERSION=1.7.10
-EMULATOR_WTF_GRADLE_PLUGIN_VERSION=0.0.10
-FLANK_VERSION=22.03.0
+ANDROID_GRADLE_PLUGIN_VERSION=7.3.0
+DETEKT_VERSION=1.22.0
+DOKKA_VERSION=1.7.20
+EMULATOR_WTF_GRADLE_PLUGIN_VERSION=0.0.12
+FLANK_VERSION=22.10.0
 FULLADLE_VERSION=0.17.4
-GRADLE_VERSIONS_PLUGIN_VERSION=0.42.0
-KTLINT_VERSION=0.46.1
-KSP_VERSION=1.7.10-1.0.6
+GRADLE_VERSIONS_PLUGIN_VERSION=0.44.0
+KTLINT_VERSION=0.48.0
+KSP_VERSION=1.7.21-1.0.8
 PROTOBUF_GRADLE_PLUGIN_VERSION=0.8.19
 RUST_GRADLE_PLUGIN_VERSION=0.9.3
 
-ANDROIDX_ANNOTATION_VERSION=1.3.0
-ANDROIDX_APPCOMPAT_VERSION=1.4.2
+ANDROIDX_ACTIVITY_VERSION=1.6.1
+ANDROIDX_ANNOTATION_VERSION=1.5.0
+ANDROIDX_APPCOMPAT_VERSION=1.5.1
+ANDROIDX_COMPOSE_COMPILER_VERSION=1.4.0-alpha02
+ANDROIDX_COMPOSE_MATERIAL3_VERSION=1.1.0-alpha02
+ANDROIDX_COMPOSE_VERSION=1.3.1
 ANDROIDX_CONSTRAINT_LAYOUT_VERSION=2.1.4
-ANDROIDX_CORE_VERSION=1.8.0
-ANDROIDX_ESPRESSO_VERSION=3.4.0
-ANDROIDX_LIFECYCLE_VERSION=2.4.1
+ANDROIDX_CORE_VERSION=1.9.0
+ANDROIDX_DATABASE_VERSION=2.2.0
+ANDROIDX_ESPRESSO_VERSION=3.5.0
+ANDROIDX_LIFECYCLE_VERSION=2.6.0-alpha03
 ANDROIDX_MULTIDEX_VERSION=2.0.1
-ANDROIDX_NAVIGATION_VERSION=2.4.2
+ANDROIDX_NAVIGATION_VERSION=2.5.3
+ANDROIDX_NAVIGATION_COMPOSE_VERSION=2.5.3
+ANDROIDX_NAVIGATION_FRAGMENT_VERSION=2.4.2
 ANDROIDX_PAGING_VERSION=2.1.2
+ANDROIDX_PROFILE_INSTALLER_VERSION=1.3.0-alpha02
 ANDROIDX_ROOM_VERSION=2.4.3
-ANDROIDX_TEST_JUNIT_VERSION=1.1.3
-ANDROIDX_TEST_ORCHESTRATOR_VERSION=1.1.0-alpha1
-ANDROIDX_TEST_VERSION=1.4.0
-ANDROIDX_UI_AUTOMATOR_VERSION=2.2.0-alpha1
+ANDROIDX_SECURITY_CRYPTO_VERSION=1.1.0-alpha04
+ANDROIDX_TEST_JUNIT_VERSION=1.1.4
+ANDROIDX_TEST_MACROBENCHMARK_VERSION=1.2.0-alpha08
+ANDROIDX_TEST_ORCHESTRATOR_VERSION=1.4.2
+ANDROIDX_TEST_VERSION=1.5.0
+ANDROIDX_TRACING_VERSION=1.2.0-alpha01
+ANDROIDX_UI_AUTOMATOR_VERSION=2.3.0-alpha01
 BIP39_VERSION=1.0.4
 COROUTINES_OKHTTP=1.0
-GOOGLE_MATERIAL_VERSION=1.6.1
-GRPC_VERSION=1.48.1
+GOOGLE_MATERIAL_VERSION=1.7.0
 GRPC_KOTLIN_VERSION=1.3.0
+GRPC_VERSION=1.51.0
```

Comment:
Have you updated the generated files with `protoc` ?

Reply:
I think they get generated automatically as part of the Gradle build, based upon the contents of src/main/proto.  See https://github.com/zcash/zcash-android-wallet-sdk/blob/7dd6d3c3429a456e6a0308925026f3e0c27dfc3b/lightwallet-client-lib/build.gradle.kts#L166

Is there something specific that you were thinking about?

Reply:
Great!
> Is there something specific that you were thinking about?


Nothing in particular, just checking!

Reply:
Could you guys please check Alex's latest changes related to the proto files update, if they meet our expectations? I will need to make similar changes in this PR, so we don't discuss it twice. ([this commit](https://github.com/zcash/zcash-android-wallet-sdk/pull/804/commits/19c072c5592496ca07ca9d6983c832b4cb01099d) and [this change](https://github.com/zcash/zcash-android-wallet-sdk/pull/804/commits/83051bdedfca6490490e1b7746c5a6f44b048680#r1085327159))